### PR TITLE
fix(view): error on missing version

### LINKED ISSUE #4964: [BUG] npm info exits with zero exit code if package@version is not found
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

If I do:

```
npm info npm@8.7.1 && echo $?
```

Then the exit code is 0, but no output from `npm info` is printed, as `8.7.1` of npm doesn't exist. I would expect `npm info` to give a non-zero exit code if the package at the specified version doesn't exist. If the package doesn't exist at all, then `npm info` does have a non-zero exit code (with logs telling you it doesn't exist)

### Expected Behavior

```
npm info npm@8.7.1
```

Would log a message saying the specified version of that package doesn't exist, and exit with a non-zero exit code, much like for non-existent packages.

### Steps To Reproduce

See above, assuming `npm@8.7.1` hasn't been published in the meantime.

### Environment

- npm: 8.7.1
- Node.js: v16.13.1
- OS Name: MacOS latest
- System Model Name: 
- npm config: no non-default configuration.

### PR DESCRIPTION
**Pull request by wraithgar**
This fixes an error in npm show. When calling npm show with a specific
version of a package that does not exist, it does not show anything and
gives a zero exit code. This has been changed: now it gives a 404 Error
similar to if the package does not exist. Can be tested with npm show
express@5.0.0 (local: node bin/npm-cli.js info express@5.0.0)

Fixes #4964

Credit: @lukaskuhn-lku

### CODE REVIEW

**Comment by wraithgar** [comment]
This is a continuation of https://github.com/npm/cli/pull/5011, I fixed the tests.

**Review by nlf — APPROVED** [review]

File: lib/commands/view.js [hunk 1]
```diff
@@ -236,6 +236,15 @@ class View extends BaseCommand {
       }
     })
 
+    // No data has been pushed because no data is matching the specified version
+    if (data.length === 0 && version !== 'latest') {
```
Comment (ljharb) [inline lib/commands/view.js:240]: is an asterisk normalized to latest?

**Review by ljharb — COMMENTED** [review]
What about a dist-tag that doesn’t exist?

File: lib/commands/view.js (hunk 1 above)
Comment (wraithgar) [reply lib/commands/view.js:240]: No, but this bug isn't currently present if you query for `pkg@*` you already get a proper error.

**Comment by wraithgar** [comment]
> What about a dist-tag that doesn’t exist?

This PR also fixes that situation

```sh
~/D/n/cli (gar/5011|✔) $ node . view lodash@asdf
npm ERR! code E404
npm ERR! 404 No match found for version asdf
npm ERR! 404 
npm ERR! 404  'lodash@asdf' is not in this registry.
```

**Comment by MarshallOfSound** [comment]
@wraithgar Probably not worth a new issue but I wanted to note that this caused a ~catastrophic~ relatively bad release infrastructure failure for the Electron project.

We have since recovered and worked around the issue and ensured the `npm` is correctly pinned in our deployment environment but this was IMO a drastic breaking change in behaviour in a patch bump of the NPM cli.

Specifically we were checking if a version had been published to npm via the CLI using:

```bash
npm show electron@1.2.3 --json
```

And assuming that "no output" meant the version had not been published and "valid output" could be parsed into the registry metadata.  In both cases since the dawn of npm the command exited cleanly, this change caused a non-zero exit code in the unpublished scenario which then propagated as an error through our release scripts causing an infinite backoff-and-retry (as we assumed a non-zero exit code meant npm was having issues).

Not sure if this is even gonna scratch on the radar, but I'd consider a change in exit code semantics for a CLI tool a breaking change and would be reserved for the next major version of npm.

**Pull request merged**